src/ndarray/ndarray.cc

```diff
@@ -614,8 +614,7 @@ NDArray &NDArray::operator/=(const real_t &src) {
 }
 
 void NDArray::Save(dmlc::Stream *strm) const {
-  // save shape
-  shape_.Save(strm);
```

Comment:
@ZihengJiang maybe consider to do a backward compatible version now by utilizing the hint?